src/graph.rs

```diff
@@ -98,6 +120,26 @@ impl Graph {
             .or_insert(Node::new(id, weight));
     }
 
+    pub fn get_num_edges(&mut self) -> usize{
+        let mut num_edges = 0;
+
+        for (_node, edges) in self.adj_list.iter_mut() {
+            num_edges += edges.len();
+        }
+
+        num_edges/2
```

Comment:
Usar um map com um sum aqui é mais performático, pois evita que seja criado um iterador mutável e a operação de atribuição também

`self.adj_list.values().map(|edges| edges.len()).sum::<usize>() / 2`

Assim ficaria mais performático mas assim, não é uma função que a gente vai usar muitas vezes, deixar da forma que está é mais "entendível" então vai da sua preferência.

Reply:
Lancei a boa craque. É uma opção, talvez mais simples, aumentar o número de arestas ao adicionar uma nova aresta, sendo a informação da quantidade de arestas presente no grafo.

Reply:
Papo reto, a gente pode ver a necessidade conforme formos desenvolvendo. No meu trab eu tive q fazer isso pois atualizar os candidatos da solução necessitava saber o número de arestas, mas se a gente n for usar da pra manter assim